### PR TITLE
Angelez/bibentries

### PR DESCRIPTION
**Pull request by geli-gel**
This is my PR for https://github.com/allenai/scholar/issues/33041 step 1: move code to mmda

In moving the code over I learned I had to figure out how the outputs will be used in SPP / stored in the annotation store, and talked w/Kyle and realized the importance of including the raw model output, which is why a `Prediction` includes `bib_entries` which contain boxes tightened around the tokens found inside of them + text and spans, as well as `original_boxes` which are raw model output detected boxes.

There are optional config variables: `BIB_ENTRY_DETECTION_PREDICTOR_SCORE_THRESHOLD` (Prediction accuracy score used to determine threshold of returned predictions) and `BIB_ENTRY_DETECTION_MIN_VILA_BIB_ROWS` (Minimum number of rows in a Bibliography VILA SpanGroup required to qualify as a Bibliography section) that can be adjusted if e2e eval shows signals we should change them.

### CODE REVIEW

File: mmda/predictors/d2_predictors/bibentry_detection_predictor.py
Comment (geli-gel) [inline mmda/predictors/d2_predictors/bibentry_detection_predictor.py:66]: @cmwilhelm This is question #1

File: mmda/predictors/d2_predictors/bibentry_detection_predictor.py
Comment (geli-gel) [inline mmda/predictors/d2_predictors/bibentry_detection_predictor.py:102]: @cmwilhelm This is question #2 - how do I learn more about how tokens exist in the Document that will be given as input? I'm assuming I can make it a requirement that the Document / Instance input contains Tokens per page somehow, but I'm struggling to understand how to match those tokens to what my bounding boxes contain. In my other code I use layoutparser's Layout.filter_by: `bib_tokens = page_tokens.filter_by(bib_block, center=True)`

**Review by cmwilhelm — CHANGES_REQUESTED** [review]
TIMO interface looks good to me. I'll defer to @kyleclo on the particulars of the mmda predictor since there's a fair amount of logic on using the vila info heuristically.

Just a couple requested changes:

1. Let's give `original_boxes` a more specific name since it's at the top-level document namespace.
2. We probably shouldn't depend on `vila_predictors` for your model. From what I can tell that's just there to allow the integration test to process the PDF before handing to your new predictor. I think we want to do that offline and generate a fixture JSON file instead.

File: ai2_internal/config.yaml
Comment (geli-gel) [reply ai2_internal/config.yaml:215]: yes you're right - I'll change the test as you suggested!

File: ai2_internal/config.yaml
Comment (geli-gel) [inline ai2_internal/config.yaml:192]: whoops

File: ai2_internal/bibentries_v1/integration_test.py
Comment (geli-gel) [reply ai2_internal/bibentries_v1/integration_test.py:1]: yep, that's how it got here, my bad!

**Comment by geli-gel** [comment]
Going to do some changes to the output Prediction after the sync today - s2airs doesn't need the text stored here anymore, so just going to save boxes only. (output will be bib_entries: List[BoxGroup] and original_boxes: List[BoxGroup]

File: ai2_internal/bibentry_detection_predictor/integration_test.py
Comment (geli-gel) [reply ai2_internal/bibentry_detection_predictor/integration_test.py:90]: I thought about making it more specific and decided against it but I think at least checking that the number of bib_entries and raw_bib_entry_boxes is the same would be a good addition

**Review by cmwilhelm — APPROVED** [review]
Thanks for the back and forth Angele. LGTM to merge after the merge conflict is resolved.

**Pull request merged**
